src/Compilers/Core/Portable/Emit/EditAndContinue/SymbolChanges.cs

```diff
@@ -51,21 +51,66 @@ protected SymbolChanges(DefinitionMap definitionMap, IEnumerable<SemanticEdit> e
 
         public DefinitionMap DefinitionMap => _definitionMap;
 
-        public ImmutableDictionary<ISymbolInternal, ImmutableArray<ISymbolInternal>> GetAllDeletedMethods()
+        public ImmutableDictionary<ISymbolInternal, ImmutableArray<ISymbolInternal>> GetAllDeletedMembers()
         {
             var builder = ImmutableDictionary.CreateBuilder<ISymbolInternal, ImmutableArray<ISymbolInternal>>();
 
-            foreach (var type in _deletedMembers)
+            foreach (var (type, deletedMembers) in _deletedMembers)
             {
-                if (GetISymbolInternalOrNull(type.Key) is { } typeSymbol)
+                if (GetISymbolInternalOrNull(type) is not { } typeSymbol)
                 {
-                    builder.Add(typeSymbol, ToInternalSymbolArray(type.Value));
+                    continue;
                 }
+
+                var internalSymbols = GetDeletedMemberInternalSymbols(deletedMembers, includeMethods: true, includeProperties: true, includeEvents: true);
+
+                builder.Add(typeSymbol, internalSymbols);
+
             }
 
             return builder.ToImmutable();
         }
 
+        private ImmutableArray<ISymbolInternal> GetDeletedMemberInternalSymbols(ISet<ISymbol> deletedMembers, bool includeMethods, bool includeProperties, bool includeEvents)
+        {
+            var internalSymbols = ArrayBuilder<ISymbolInternal>.GetInstance();
+
+            foreach (var symbol in deletedMembers)
+            {
+                if (GetISymbolInternalOrNull(symbol) is { } internalSymbol)
+                {
+                    if (includeProperties &&
+                        symbol is IMethodSymbol { AssociatedSymbol: IPropertySymbol propertySymbol } &&
+                        (propertySymbol.GetMethod is null || propertySymbol.GetMethod == symbol))
+                    {
+                        var internalPropertySymbol = GetISymbolInternalOrNull(propertySymbol);
+                        if (internalPropertySymbol is not null)
+                        {
+                            internalSymbols.Add(internalPropertySymbol);
+                        }
+                    }
+
+                    if (includeEvents &&
+                        symbol is IMethodSymbol { AssociatedSymbol: IEventSymbol eventSymbol } &&
+                        eventSymbol.AddMethod == symbol)
+                    {
+                        var internalEventSymbol = GetISymbolInternalOrNull(eventSymbol);
+                        if (internalEventSymbol is not null)
+                        {
+                            internalSymbols.Add(internalEventSymbol);
+                        }
+                    }
+
+                    if (includeMethods)
+                    {
+                        internalSymbols.Add(internalSymbol);
```

Comment:
>internalSymbol [](http://example.com/codeflow?start=44&length=14)

Are all symbols in deletedMembers methods?

Reply:
Yes, but it's purely driven by what the analyzer reports `SemanticEditKind.Delete` for so field will presumably end up there too. Will add a check